src/actions/firebase/firebase.ts

```diff
@@ -0,0 +1,158 @@
+  import * as Hub from "../../hub"
+  import * as admin from 'firebase-admin';
+  import * as app from 'firebase-admin/app';
+
+  export class FirebaseAction extends Hub.Action {
+
+    name = "firebase"
+    label = "Firebase"
+    iconName = "firebase/firebase.png"
+    description = "Use firebase to send push notifications to mobile."
+    storageBucket = process.env.FIREBASE_BUCKET
+    contentType = "image/jpeg"
+    notificationOptions = {
+      priority: "high",
+      timeToLive: 60 * 60 * 24,
+      contentAvailable: true,
+      mutableContent: true
+    }
+    static firebaseAdmin: app.App
+    params = []
+    supportedActionTypes = [Hub.ActionType.Query]
+    supportedFormats = [Hub.ActionFormat.JsonDetail]
+    supportedFormattings = [Hub.ActionFormatting.Unformatted]
+    supportedVisualizationFormattings = [Hub.ActionVisualizationFormatting.Noapply]
+
+    static setFirebaseClient() {
+      if (!FirebaseAction.firebaseAdmin) {
+        FirebaseAction.firebaseAdmin = admin.initializeApp({
```

Comment:
question on the way we are initialising firebaseAdmin object
- if some env variable like Private key is changed then we would need to restart action hub for it to get picked up as per current code.. is that expected and how other config changes work in action hub ?
- can the firebaseAdmin go to a `bad` state (don't know much about it) and in that case how will it get re-created.

Reply:
This is expected for restarts to be neccessary on env changes. Env vars are loaded at container startup.